pkg/repository/manager.go

```diff
@@ -89,14 +90,34 @@ type Manager struct {
 }
 
 // LoadRepository loads information about a single local repository
-func (m *Manager) LoadRepository(name string) (*Repository, error) {
+func (m *Manager) LoadRepository(name string, t RepositoryType) (*Repository, error) {
 	// check repo cache
 	repo, err := m.getRepository(name)
 	if err == nil {
 		return repo, nil
 	}
 
+	switch t {
+	case RepositoryTypeCore:
+		if !util.InStrings(name, m.cfg.Repositories.Core) {
+			return nil, ErrUnconfiguredRepository
+		}
+	case RepositoryTypeController:
+		if !util.InStrings(name, m.cfg.Repositories.Services) {
+			return nil, ErrUnconfiguredRepository
+		}
+	}
+
+	return m.AddRepository(name, t)
+}
+
+// AddRepository oads information about a single local repository and adds it to the cache.
+func (m *Manager) AddRepository(name string, t RepositoryType) (*Repository, error) {
 	repoName := name
+	// controller repositories should always have a '-controller' suffix
+	if t == RepositoryTypeController {
+		repoName = fmt.Sprintf("%s-controller", name)
+	}
```

Comment:
This is duplicated across the `NewRepository` method. Should we just be calling `NewRepository` here, instead?

Reply:
Yuppp ++